OptimizelySDKCore/OptimizelySDKCore/Optimizely.m

```diff
@@ -628,6 +668,7 @@ - (void)handleErrorLogsForActivateUser:(NSString *)userId
                                      userInfo:errorDictionary];
     [self.errorHandler handleError:error];
     [self.logger logMessage:logMessage withLevel:OptimizelyLogLevelError];
+    return error;
 }
```

Comment:
The handleErrorLogsForActivateUser:experiment: itself calls
```
     [self.errorHandler handleError:error];
```

The three locations which call handleErrorLogsForActivateUser:experiment:
may themselves execute
```
            callback(error);
```

Essentially, doesn't this mean that a single error can be processed by two different
error handlers: self.errorHandler and callback ?

We could think about how design patterns like "Single responsibility principle"
https://en.wikipedia.org/wiki/Single_responsibility_principle and more relevant
"Chain-of-responsibility pattern"
https://en.wikipedia.org/wiki/Chain-of-responsibility_pattern
could apply here.